### PR TITLE
AZP: Add wire-compat check for perf caps

### PR DESCRIPTION
**Pull request by brminich**
## What
Add comparision of transport performance charasteristics reported by `ucx_info -d`

## Why
Wire-compatibility check. Diffferent values of the same perf capability between different UCX versions may result in ep reconfiguration error

### CODE REVIEW

File: buildlib/tools/check_tls_perf_caps.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+# Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# See file LICENSE for terms.
+#
+
+old_out=$(${UCX_LEGACY_PATH}/bin/ucx_info -d)
+new_out=$(${UCX_PR_PATH}/bin/ucx_info -d)
+
+res=true
+for tl_name in `echo "${old_out}" | grep Transport | awk '{print $3}'`
```
Comment (tvegas1) [inline buildlib/tools/check_tls_perf_caps.sh:12]: detail: maybe `$()` instead of `` to match others?

Comment (brminich) [reply buildlib/tools/check_tls_perf_caps.sh]: fixed in place as pr is small

**Review by tvegas1 — DISMISSED** [review]
would it make sense to add some amount of logging. anybody opening test result could quickly check if all bash processing makes sense

**Review by tvegas1 — DISMISSED** [review]

**Review by tvegas1 — APPROVED** [review]

**Comment by brminich** [comment]
@Alexey-Rivkin can you please also review?

**Review by Alexey-Rivkin — DISMISSED** [review]
Minor: Shellcheck suggests adding more double quotes for certain vars.

File: buildlib/pr/wire_compat.yml [hunk 2]
```diff
@@ -65,6 +65,7 @@ jobs:
         inputs:
           artifactName: $(ucx_current)
           downloadPath: $(System.DefaultWorkingDirectory)
+      - bash: chmod u+rwx $(System.DefaultWorkingDirectory) -R
```
[the remark this answers is not in the document]
Comment (brminich) [reply buildlib/pr/wire_compat.yml:68]: failed to run `ucx_info` and noticed we grant permissions this way in several places
What is you suggestion?

**Review by yosefe — APPROVED** [review]

**Review by yosefe — APPROVED** [review]

**Comment by yosefe** [comment]
@brminich pls squash

**Pull request merged**